File: pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>


void Pyramid(int i);

int main(void)
{
    bool badInput = true;
    do
    {
        int height = get_int("Height: "); // Obtain user input

        if (height >= 1 && height <= 8) // check if height is not less than 1 or not greater than 8
        {
            for (int i = 0; i < height; i++)
            {
                for (int k = height; k > i + 1; k--)
                {
                    printf(" ");
                }
                Pyramid(i);
                printf("  ");
                Pyramid(i);
                printf("\n");
            }
            badInput = false; // if height is not less than 1 or not greater than 8
        }
        else
        {
            badInput = true; // if height is less than 1 or greater than 8
        }
    }
    while (badInput);
}

void Pyramid(int i) // Fuction that print pyramid
{
    for (int j = 0; j < i + 1; j++)
    {
        printf("#");
    }
}

File: pset6/mario/more/mario.py
from cs50 import get_int


def main():
    # Repeat until user enter positive number not greater than 8
    while True:
        height = get_int("Height: ")
        if height >= 1 and height <= 8:
            break
    # Call fuction pyramid
    pyramid(height, 0)

# Print Pyramid


def pyramid(h, s):
    if h != 0:
        # Hard to explain
        pyramid(h-1, s+1)
        # Print blocks
        print(" " * s + "#" * h + "  " + "#" * h)


main()

